doctr/datasets/datasets/pytorch.py

```diff
@@ -20,6 +20,16 @@
 class AbstractDataset(_AbstractDataset):
     def _read_sample(self, index: int) -> Tuple[torch.Tensor, Any]:
         img_name, target = self.data[index]
+
+        # Check target
+        if isinstance(target, dict):
+            assert "boxes" in target, "Target should contain 'boxes' key"
+            assert "labels" in target, "Target should contain 'labels' key"
+        else:
+            assert isinstance(target, str) or isinstance(
+                target, np.ndarray
+            ), "Target should be a string or a numpy array"
```

Comment:
Considering that the target is unchanged since the constructor was run, that should be moved to the constructor (because otherwise it increases data reading latency)

Reply:
you mean to check it inside every dataset ? otherwise we have only access inside _read_sample and __getitem__ !? And i don't think it changes the runtime noticeably